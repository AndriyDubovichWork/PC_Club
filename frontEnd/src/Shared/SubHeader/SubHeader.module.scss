.backGroundPC {
  position: absolute;
  left: 0;
  top: 0;

  width: 100%;
  height: 40vh;

  z-index: -1;

  background: center center fixed no-repeat url('./assets/backGroundPC.jpg');
  background-size: cover;
  filter: opacity(0.3);
}
.Text {
  margin: 10vh 0;
  .Title {
    font-weight: 800;
    font-size: 2rem;
  }

  .Link {
    font-weight: 600;
    color: #000;
    text-decoration: none;
  }
  .Link:hover {
    color: #ff4d30;
  }
}
