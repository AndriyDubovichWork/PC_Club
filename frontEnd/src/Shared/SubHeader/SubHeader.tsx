import React from 'react';
import { Breadcrumbs } from '@mui/material';
import { NavLink, useLocation } from 'react-router-dom';
import style from './SubHeader.module.scss';

type SubHeaderPropsT = {
  title: string;
};

const SubHeader = ({ title }: SubHeaderPropsT) => {
  const location = useLocation();

  return (
    <>
      <img className={style.backGroundPC} loading='lazy' />

      <div className={style.Text}>
        <h1 className={style.Title}>{title}</h1>

        <Breadcrumbs aria-label='breadcrumb'>
          <NavLink to='/PC_Club' className={style.Link}>
            Home
          </NavLink>
          <NavLink to={location.pathname} className={style.Link}>
            {title}
          </NavLink>
        </Breadcrumbs>
      </div>
    </>
  );
};

export default SubHeader;
