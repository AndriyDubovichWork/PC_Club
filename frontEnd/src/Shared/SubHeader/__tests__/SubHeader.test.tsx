import { render, screen, cleanup } from '@testing-library/react';
import SubHeader from '../SubHeader';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('SubHeader match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/about']}>
        <Routes>
          <Route path='/PC_Club/about' element={<SubHeader title='hello' />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('SubHeader contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/about']}>
      <Routes>
        <Route path='/PC_Club/about' element={<SubHeader title='hello' />} />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('Home')).toBeInTheDocument();
  expect(screen.getByText('/')).toBeInTheDocument();
});
