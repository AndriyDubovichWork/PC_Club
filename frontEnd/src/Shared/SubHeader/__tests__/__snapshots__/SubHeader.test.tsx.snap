// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader match snapshot 1`] = `
Array [
  <img
    className="backGroundPC"
    loading="lazy"
  />,
  <div
    className="Text"
  >
    <h1
      className="Title"
    >
      hello
    </h1>
    <nav
      aria-label="breadcrumb"
      className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          className="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            className="Link active"
            href="/PC_Club"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          aria-hidden={true}
          className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          className="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            className="Link active"
            href="/PC_Club/about"
            onClick={[Function]}
          >
            hello
          </a>
        </li>
      </ol>
    </nav>
  </div>,
]
`;
